examples/website/openstreet-map/app.js

```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+
+import {StaticMap} from 'react-map-gl';
+import DeckGL, {TileLayer, BitmapLayer} from 'deck.gl';
+import {loadImage} from '@loaders.gl/core';
+
+// Set your mapbox token here
+const MAPBOX_TOKEN = process.env.MapboxAccessToken; // eslint-disable-line
+
+export const INITIAL_VIEW_STATE = {
+  latitude: 47.65,
+  longitude: 7,
+  zoom: 4.5,
+  maxZoom: 16,
+  pitch: 50,
+  bearing: 0
+};
+
+const s = 'c';
+
+// https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Lon..2Flat._to_tile_numbers_2
+function tile2latLng(x, y, z) {
+  const lng = (x / Math.pow(2, z)) * 360 - 180;
+  const n = Math.PI - (2 * Math.PI * y) / Math.pow(2, z);
+  const lat = (180 / Math.PI) * Math.atan(0.5 * (Math.exp(n) - Math.exp(-n)));
+  return [lng, lat];
+}
+
+function tileIndexToBounds({x, y, z}) {
+  const [west, north] = tile2latLng(x, y, z);
+  const [east, south] = tile2latLng(x + 1, y + 1, z);
+
+  return [west, south, east, north];
+}
+
+export class App extends Component {
+  constructor(props) {
+    super(props);
+    this._getTileData = this._getTileData.bind(this);
+  }
+
+  _renderLayers() {
+    return [
+      new TileLayer({
+        getTileData: this._getTileData,
+        renderSubLayers: props => {
+          const layerId = `bitmap-${props.id}`;
+          props.data.then(data => {
+            this.setState({
+              [layerId]: {
+                image: data.image,
+                bounds: data.bounds
+              }
+            });
+          });
+
+          return new BitmapLayer({id: layerId, ...this.state[layerId]});
+        }
```

Comment:
From my understanding, `renderSubLayers` callback makes sense when the SubLayer expects `data` field, deck.gl Layer auto resolves the `data` promise and feed resolved data to that SubLayer. 

But `BitmapLayer` does not rely on data, and it uses `image` and `bounds` to place the image. So I have to manipulate the react state to trigger the layer re-render when data resolved.